targets/HUDL/main.cpp

```diff
@@ -6,185 +6,64 @@
-// @param: col_low : the last four bits of the column write
-// @param: data : the data value to write
-void HUDL::drive_pixel(unsigned char page, unsigned char col_up, unsigned char col_low, unsigned char data ) {
-    comm_write(0x40); //line to start writing on (0 -> 64) moves set bits with it DO NOT CHANGE 
-    comm_write(0xB0+ page); //writes the page address (4 bits, 8 rows selcted by values 0-7 ) 
-    comm_write(0x10 + col_up); //writes the first 4 bits of the column select (out of 8 bits)
-    comm_write(0x00 + col_low); //writes the second 4 bits of the column select (out)
-
-    data_write(data); //writes 8 vertical bits based on value between 0-255 based on bits set ex: 01001100 is       |WHITE|
-                      //                                                                                            |BLACK|
-                      //                                                                                            |WHITE|
-                      //                                                                                            |WHITE|
-                      //                                                                                            |BLACK|
-                      //                                                                                            |BLACK|
-                      //                                                                                            |WHITE|
-                      //                                                                                            |WHITE|             
-  }
-
-
-// clears the LCD screen
-// @param: lcd_string : 
-void HUDL::ClearLCD(unsigned char *lcd_string)
-{
-    unsigned int i,j;
-    unsigned char page = 0xB0;
-    comm_write(0xAE);          //Display OFF
-    comm_write(0x40);         //Display start address + 0x40
-    for(i=0;i<8;i++){       //64 pixel display / 8 pixels per page = 8 pages
-        comm_write(page);       //send page address
-        comm_write(0x10);       //column address upper 4 bits + 0x10
-        comm_write(0x00);       //column address lower 4 bits + 0x00
-    for(j=0;j<128;j++){     //128 columns wide
-        data_write(0x00);    //write clear pixels
-        lcd_string++; 
-        }
-        page++;         //after 128 columns, go to next page
-        }
-    comm_write(0xAF);   
-}
-
-
-// initializes LCD for use
-void HUDL::init_LCD()  {
-    comm_write(0xA0);   // ADC select 
-    comm_write(0xAE);   // Display OFF
-    comm_write(0xC8);   // COM direction scan 
-    comm_write(0xA2);   // LCD bias set
-    comm_write(0x2F);   // Power Control set
-    comm_write(0x26);   // Resistor Ratio Set 
-    comm_write(0x81);   // Electronic Volume Command (set contrast) Double Btye: 1 of 2
-    comm_write(0x11);   // Electronic Volume value (contrast value) Double Byte: 2 of 2
-    comm_write(0xAF);   // Display ON
-}
-
-
+const uint32_t SPI_SPEED = SPI_SPEED_4MHZ; // 4MHz
```

Comment:
Why const instead of constexpr?

Reply:
SPI_SPEED_4MHZ is just a macro defined integer meaning the value is known at compile time. Therefore SPI_SPEED would still be a compile time constant.

Reply:
That makes sense, but `constexpr` is the keyword used for compile-time constants, not `const`. I read some info here: https://stackoverflow.com/questions/13346879/const-vs-constexpr-on-variables